src/Microsoft.AspNet.OData.Shared/ODataIdContainer.cs

```diff
@@ -0,0 +1,29 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataIdContainer.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Interface for ODataId container. It will be used by POCO objects as well as Delta{TStructuralType}.
+    /// </summary>
+    public interface IODataIdContainer
+    {
+        /// <summary>
+        /// Gets or set the ODataId path string.
+        /// </summary>
+        string ODataId { set; get; }
+    }
+
+    /// <summary>
+    /// Default implementation of IOdataIdContainer
+    /// </summary>
+    public class ODataIdContainer : IODataIdContainer
```

Comment:
I still have reservations about this class and interface. I don't have a suggestion, nor is this a blocker for me. But not convinced why we need it, why it needs to be a new public class, why it needs a public interface, etc. Do we expect customers to create their own implementations? Do you have a sample implementation or use-case you can share?

Reply:
users do not need to create their own implementation but they need to have a type of this property in their `POCO` classes when they want to reference another type in their types.